File: jambeth/jAmbeth.Merge/src/de/osthus/ambeth/cache/ICacheModification.java
package de.osthus.ambeth.cache;

import de.osthus.ambeth.threading.IBackgroundWorkerDelegate;

public interface ICacheModification
{
	boolean isInternalUpdate();

	void setInternalUpdate(boolean internalUpdate);

	boolean isActive();

	void setActive(boolean active);

	boolean isActiveOrFlushing();

	void queuePropertyChangeEvent(IBackgroundWorkerDelegate task);
}

File: ambeth/Ambeth.Merge/ambeth/cache/ICacheModification.cs
using System;
using De.Osthus.Ambeth.Model;
using System.ComponentModel;
using De.Osthus.Ambeth.Threading;

namespace De.Osthus.Ambeth.Cache
{
    public interface ICacheModification
    {
        bool InternalUpdate { get; set; }

        bool Active { get; set; }

        bool ActiveOrFlushing { get; }
        
        void QueuePropertyChangeEvent(IBackgroundWorkerDelegate task);
    }
}
